### PR TITLE
Fix openapi

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Fixes https://github.com/DirectXMan12/k8s-prometheus-adapter/issues/329

Update custom-metrics-apiserver to include https://github.com/kubernetes-sigs/custom-metrics-apiserver/pull/75 and update openapi generation task to keep schema in sync with k8s version.

### CODE REVIEW

**Comment by s-urbaniak** [comment]
leaving the final review to @dgrisonnet :-)

File: Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ endif
 all: $(OUT_DIR)/$(ARCH)/adapter
 
 src_deps=$(shell find pkg cmd -type f -name "*.go")
-$(OUT_DIR)/%/adapter: $(src_deps)
+$(OUT_DIR)/%/adapter: pkg/api/generated/openapi/zz_generated.openapi.go $(src_deps)
```
Comment (dgrisonnet) [inline Makefile:36]: this file should already be part of the `src_deps` variable

File: Makefile [hunk 2]
```diff
@@ -80,7 +80,6 @@ go-mod:
 
 verify: verify-gofmt go-mod test
 
-pkg/api/generated/openapi/zz_generated.openapi.go:
-	GO111MODULE=off go get -d k8s.io/kube-openapi || true
+pkg/api/generated/openapi/zz_generated.openapi.go: go.mod go.sum
 	rm -rf pkg/api/generated/openapi
 	cd $(OPENAPI_PATH) && go run ./cmd/openapi-gen/openapi-gen.go --logtostderr -i k8s.io/metrics/pkg/apis/custom_metrics,k8s.io/metrics/pkg/apis/custom_metrics/v1beta1,k8s.io/metrics/pkg/apis/custom_metrics/v1beta2,k8s.io/metrics/pkg/apis/external_metrics,k8s.io/metrics/pkg/apis/external_metrics/v1beta1,k8s.io/metrics/pkg/apis/metrics,k8s.io/metrics/pkg/apis/metrics/v1beta1,k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/api/resource,k8s.io/apimachinery/pkg/version,k8s.io/api/core/v1 -p github.com/directxman12/k8s-prometheus-adapter/pkg/api/generated/openapi -O zz_generated.openapi -r /dev/null
```
Comment (dgrisonnet) [inline Makefile:85]: while we are at it, would you mind updating this command to make it gopath independent? Like we did here: https://github.com/kubernetes-sigs/custom-metrics-apiserver/pull/76/commits/97bc9e2bbd45c40a6de998965e713897099ef9ce

**Review by dgrisonnet — COMMENTED** [review]
I just have a few nit comments. Thank you for taking the time to look into this issue @johanneswuerbach.

File: Makefile (hunk 2 above)
Comment (johanneswuerbach) [reply Makefile]: Changed

File: Makefile (hunk 1 above)
Comment (johanneswuerbach) [reply Makefile]: Done

**Comment by s-urbaniak** [comment]
lgtm once green, thank you everybody for tackling this one! Once merged, I will release 0.8.2 :-)

**Pull request merged**